js/ui/notificationDaemon.js

```diff
@@ -740,6 +745,11 @@ const GtkNotificationDaemonAppSource = new Lang.Class({
         else
             this.pushNotification(notification);
 
+        if (oldNotification) {
+            oldNotification.disconnect(oldNotificationDestroyId);
+            oldNotification.destroy();
+        }
+
```

Comment:
Is there any reason why we can't disconnect and destroy the old notification earlier, before we add the new notification?